lumicks/pylake/kymotracker/kymoline.py

```diff
@@ -416,3 +416,49 @@ def fit_binding_times(self, n_components):
         return _kinetic_mle_optimize(
             n_components, dwelltimes_sec, min_observation_time, max_observation_time
         )
+
+    def _histogram_binding_events(self, kind, bins=10):
+        """Make histogram of bound events.
+
+        Parameters
+        ----------
+        kind : {'binding', 'all'}
+            Type of events to count. 'binding' counts only the first position of each track whereas
+            'all' counts positions of all time points in each track.
+        bins : int or sequence of scalars
+            Definition of the histogram bins; passed to `np.histogram()`. If `int`, defines the total
+            number of bins. If sequence of scalars, defines the edges of the bins.
+        """
+        if kind == "all":
+            slc = slice(None)
+        elif kind == "binding":
+            slc = slice(0, 1)
+        else:
+            raise ValueError(f"`kind` argument '{kind}' must be 'all' or 'binding'.")
+        binds = np.hstack([track.position[slc] for track in self])
+
+        counts, edges = np.histogram(binds, bins=bins)
```

Comment:
We already discussed this briefly, but just so it isn't forgotten.

It would be good to specify the `range` parameter to make sure the `extents` correspond to the kymograph's `extents`. This will make the binning with a set number of bins reproducible, no matter where the lines are actually tracked.

This can be useful if we're for example looking at a few slices of the same kymo, and want to compare binding histograms. If we don't specify the `range`, the bin edges will depend on the lines tracked (using the minimum and maximum of the found line positions). Ideally they shouldn't, so histograms for a particular kymo are directly comparable.

In the future, we could consider maybe an optional argument for the range behaviour (`full` (default), `auto` or a manually specified range for instance), but for now this `full` behaviour is probably sufficient.

Reply:
good point! I also updated the documentation to specifically address this point